### PR TITLE
Add infinite history and search history

### LINKED ISSUE #2877: [Feature Request] Increase max Quantity of History to be show
I sometimes read between 20 mangas a day lol during this quarantine so the history is very low, can we make it longer by chose?

### PR DESCRIPTION
**Pull request by jobobby04**
Added slightly modified infinite history and search history from AZ/SY

Closes #2877

### CODE REVIEW

File: app/src/main/java/eu/kanade/tachiyomi/data/database/queries/RawQueries.kt [hunk 1]
```diff
@@ -66,8 +66,36 @@ fun getRecentMangasQuery() =
...
+    LIMIT 25 OFFSET $offset
+"""
+
+/**
+ * Query to get the recently read chapters of manga from the library up to a date.
+ * The max_last_read table contains the most recent chapters grouped by manga
+ * The select statement returns all information of chapters that have the same id as the chapter in max_last_read
+ * and are read after the given time period
+ */
+fun getRecentMangasLimitQuery(limit: Int = 25, search: String = "") =
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/data/database/queries/RawQueries.kt:80]: Couldn't you just have a single function with both limit and offset?

File: app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt [hunk 2]
```diff
@@ -29,26 +28,67 @@ class HistoryPresenter : BasePresenter<HistoryController>() {
      */
     val db: DatabaseHelper by injectLazy()
 
+    private var lastCount = 25
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt:31]: Ought to have a constant for this that's reused for the raw query default too.

File: app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt [hunk 3]
```diff
@@ -29,26 +28,67 @@ class HistoryPresenter : BasePresenter<HistoryController>() {
...
 
     /**
      * Get recent manga observable
      * @return list of history
      */
-    fun getRecentMangaObservable(): Observable<List<HistoryItem>> {
+    /**
+     * Get recent manga observable
+     * @return list of history
+     */
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt:60]: Remove duplicated JavaDoc

File: app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt [hunk 4]
```diff
@@ -29,26 +28,67 @@ class HistoryPresenter : BasePresenter<HistoryController>() {
      */
     val db: DatabaseHelper by injectLazy()
 
+    private var lastCount = 25
+    private var lastSearch = ""
```
Comment (arkon) [inline app/src/main/java/eu/kanade/tachiyomi/ui/recent/history/HistoryPresenter.kt:32]: I'm not sure I understand what this is for?

**Comment by jobobby04** [comment]
The latest commit should resolve most of the comments

**Pull request merged**